src/auxiliary.jl

```diff
@@ -8,27 +8,35 @@
 
 ## Auxiliary function ##
 
-function check_taylor1_order!{T<:Number}(coeffs::Array{T,1}, order::Int)
+"""
+    resize_coeffs1!{T<Number}(coeffs::Array{T,1}, order::Int)
```

Comment:
I moved part of the constructor of `Taylor1` to this function to simplify the code. I'm not sure how to answer the question, since this function resizes the coefficients if you ask for a larger order. For instannce, you may call `Taylor1([0,1],10)` and at some point this function is called to resize the coefficients to size 11.